invenio_app_ils/ill/loaders/jsonschemas/patron_loan_extension_actions.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Schema for Borrowing Request accept extension action."""
+
+import arrow
+from flask_babelex import lazy_gettext as _
+from invenio_circulation.records.loaders.schemas.json import DateString
+from invenio_rest.serializer import BaseSchema as InvenioBaseSchema
+from marshmallow import EXCLUDE, ValidationError, fields, pre_load, validates
+
+from invenio_app_ils.circulation.loaders.schemas.json.base import \
+    transaction_location_pid_validator
+from invenio_app_ils.ill.api import BorrowingRequest
+
+
+def validate_statuses(record):
+    """Validate that the extension action can be performed."""
+    if record["status"] != "ON_LOAN":
+        raise ValidationError(
+            "This interlibrary loan is currently not on loan."
+        )
+
+    ext_status = (
+        record.get("patron_loan", {}).get("extension", {}).get("status")
+    )
+    is_valid = ext_status in BorrowingRequest.EXTENSION_STATUSES
+    if ext_status and not is_valid:
```

Comment:
Why not directly `if ext_status in BorrowingRequest.EXTENSION_STATUSES` and skip the `is_valid`

Reply:
Because in normal cases `ext_status` is non-existing: the check make sure that, if defined, that must be valid.
Extension statuses are: [`PENDING`, `DECLINED`]. The absence of it means that the extension is not requested.

The initial design of this feature had an extra state `NOT_REQUESTED`, but this will imply that any new `BorrowingRequest` record will have, by default, newly created a:
```
...
patron_loan: {
  extension: {
    status: 'NOT_REQUESTED'
  }
  pid: None
}
...
```
IMHO, this is polluting very much records, so I removed.